internal/services/storage/storage_share_file_resource.go

```diff
@@ -112,7 +136,19 @@ func resourceStorageShareFileCreate(d *pluginsdk.ResourceData, meta interface{})
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	storageShareId, err := shares.ParseShareID(d.Get("storage_share_id").(string), storageClient.StorageDomainSuffix)
+	var (
+		storageShareId *shares.ShareId
+		err            error
+	)
+	if features.FivePointOh() {
+		storageShareId, err = shares.ParseShareID(d.Get("storage_share_url").(string), storageClient.StorageDomainSuffix)
+	} else {
+		storageShareURL := d.Get("storage_share_url")
+		if storageShareURL == "" {
+			storageShareURL = d.Get("storage_share_id")
+		}
+		storageShareId, err = shares.ParseShareID(storageShareURL.(string), storageClient.StorageDomainSuffix)
+	}
 	if err != nil {
 		return err
 	}
```

Comment:
As above, can we keep the normal form, and error check at the point err is populated rather than outside the conditional.

Reply:
Done